shared/health/src/main/java/io/pravega/shared/health/impl/HealthDaemonImpl.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health.impl;
+
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.HealthDaemon;
+import io.pravega.shared.health.HealthService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+@RequiredArgsConstructor
+public class HealthDaemonImpl implements HealthDaemon {
```

Comment:
Make it extend from `AbstractScheduledService`. That will take care of starting, stopping, and your executor and futures. Check out `CacheManager.java` or `Reporter.java` for how this is used elsewhere.

Also make your class implement `AutoCloseable` (do that in the interface) and make sure your service stops when `close` is invoked. See those other classes as examples.